graylog2-server/src/main/java/org/graylog2/filters/RulesFilter.java

```diff
@@ -16,90 +16,79 @@
         this.filterService = filterService;
+        this.scheduler = scheduler;
 
-        currentFilterSet = Sets.newHashSet();
-        cache = CacheBuilder.newBuilder()
-                .expireAfterWrite(1, TimeUnit.SECONDS)
-                .build();
-        privateSession = rulesEngine.createPrivateSession();
+        loadRules();
+
+        // TODO: This class needs lifecycle management to avoid leaking objects in the EventBus
+        serverEventBus.register(this);
     }
 
     @Override
     public boolean filter(Message msg) {
-        final Set<FilterDescription> filters;
-        try {
-            filters = cache.get("filters", new Callable<Set<FilterDescription>>() {
-                @Override
-                public Set<FilterDescription> call() throws Exception {
-                    // TODO this should be improved by computing the difference between the filter sets
-                    // most of the time nothing changes at all.
-                    final Set<FilterDescription> newFilters = filterService.loadAll();
-                    final Sets.SetView<FilterDescription> difference =
-                            Sets.symmetricDifference(currentFilterSet, newFilters);
-                    if (difference.isEmpty()) {
-                        // there wasn't any change, simply return the current filter set
-                        LOG.debug("Filter sets are identical, not updating rules engine.");
-                        return currentFilterSet;
-                    }
-
-                    // something changed, we simply update everything, not trying to do the minimal changes yet
-                    // should this become too expensive we need to do something smarter
-                    LOG.debug("Updating rules engine, filter sets differ: {}", difference);
-                    // retract all current filter facts
-                    for (FilterDescription filterDescription : currentFilterSet) {
-                        privateSession.deleteFact(filterDescription);
-                    }
-                    // state all new filter facts
-                    for (FilterDescription filterDescription : newFilters) {
-                        privateSession.insertFact(filterDescription);
-                    }
-                    currentFilterSet.clear();
-                    // remember currently stated facts
-                    currentFilterSet.addAll(newFilters);
-                    return currentFilterSet;
-                }
-            });
-        } catch (ExecutionException ignored) {
-            return false;
-        }
-
         // Always run the rules engine to make sure rules from the external rules file will be run.
-        privateSession.evaluate(msg, true);
+        privateSession.get().evaluate(msg, true);
 
         // false if not explicitly set to true in the rules.
         return msg.getFilterOut();
     }
 
+    @Subscribe
+    public void handleRulesUpdate(FilterDescriptionUpdateEvent ignored) {
```

Comment:
Adding `@SuppressWarnings("unused")` would disable the compiler warnings (and make the IDE happy).